### PR TITLE
fix issue: getting metrics failed

### PR DESCRIPTION
**Pull request by QiaolingTang**
When checking the CI testing results, the cases the curl metrics always failed, I open this PR to use another pod to execute the curl command. 

@anpingli @akostadinov  PTAL, thanks.

### CODE REVIEW

**Comment by QiaolingTang** [comment]
Log: https://url.corp.redhat.com/4dcb4e0

**Comment by anpingli** [comment]
@QiaolingTang  LGTM. By the ways,  as logging forward feature will be launched later. it is better to run in the fluentd pod by default

**Comment by QiaolingTang** [comment]
> @QiaolingTang LGTM. By the ways, as logging forward feature will be launched later. it is better to run in the fluentd pod by default

OK, for ES metrics, run command in the ES pod, for fluentd metrics, run command in the fluentd pod.

File: features/step_definitions/logging.rb [hunk 1]
```diff
@@ -301,12 +301,27 @@
...
+    step %Q/a pod becomes ready with labels:/, table(%{
+      | cluster-name=elasticsearch,component=elasticsearch |
+    })
+    container_name = "elasticsearch"
+  when "fluentd"
+    step %Q/a pod becomes ready with labels:/, table(%{
+      | logging-infra=fluentd |
+    })
+    container_name = "fluentd"
+  end
```
Comment (liangxia) [inline features/step_definitions/logging.rb]: Maybe make it more readable like below ?
```
  case pod_type
  when "CLO"
    selector = "name=cluster-logging-operator"
    container_name = "cluster-logging-operator"
  when "ES"
    selector = "cluster-name=elasticsearch,component=elasticsearch"
    container_name = "elasticsearch"
  when "fluentd"
    selector = "logging-infra=fluentd"
    container_name = "fluentd"
  end
  step %Q/a pod becomes ready with labels:/, table(%{
    | #{selector} |
  })
```

Comment (QiaolingTang) [reply features/step_definitions/logging.rb]: @liangxia Good idea, thanks a lot.

**Comment by QiaolingTang** [comment]
@pruan-rht could you please help to merge this PR? Thanks.

**Pull request merged**